lib/NodeConnection.js

```diff
@@ -179,41 +189,42 @@ define(function (require, exports, module) {
      * Helper function to attempt a single connection to the node server
      */
     NodeConnection.prototype._attemptSingleConnect = function () {
-        var deferred = Promise.defer();
-        var port = null;
-        var ws = null;
-        setDeferredTimeout(deferred, CONNECTION_TIMEOUT);
+        var getRemotePromised = Promise.promisify(this.getRemotePort, this);
```

Comment:
No, but I get why you don't like it.  It doesn't seem right to call it `getRemotePromise` because it is a function that returns a promise?  Anyway, it will be changed in the next commit.  See if you like the new version.